src/olympia/bandwagon/views.py

```diff
@@ -76,10 +76,16 @@ def get_serializer_class(self):
             CollectionSerializer if not with_addons else CollectionWithAddonsSerializer
         )
 
+    def get_object(self):
+        if not hasattr(self, 'instance') and 'pk' in self.kwargs:
+            self.lookup_field = 'pk'
+        return super().get_object()
```

Comment:
Is this exposing extra functionality to the collection api endpoint itself?  iirc it was a deliberate design to only have collections accessible by slug.

Reply:
No, it's not exposed: When generating the URL for `CollectionViewSet`, we use `NestedSimpleRouter` with `lookup='user'` (so `kwargs['user_pk']` is passed) + `.register()` (so `kwargs[CollectionViewSet.lookup_field]` is passed - `lookup_field` is set to `slug` by default).

That being said, I can make it safer - I can pass a custom `action` and only do the `lookup_field = 'pk'` bit if we are using that action - making it truly impossible to use that path even if we were to accidentally pass `'pk'` in `self.kwargs`.

Reply:
I don't want to add a whole bunch of extra complexity for a scenario that can't happen... but it was unexpected that we'd be adding another way to lookup instances to CollectionViewset, for abuse reports.  Up to you 🤷‍♂️

Reply:
29c454ea38ae82c026cc9b90ac6fc3a5290c6c5e simplifies the handling and makes it harder to accidentally end up in this scenario, while still allowing reports by pk